storage/rocksdb/ha_rocksdb.cc

```diff
@@ -2985,6 +3116,54 @@ class Rdb_snapshot_status : public Rdb_tx_list_walker {
                               tx->get_lock_count(), tx->get_write_count());
     }
   }
+
+  void populate_deadlock_buffer() {
+    auto dlock_buffer = rdb->GetDeadlockInfoBuffer();
+    m_data += "----------LATEST DETECTED DEADLOCKS----------\n";
+
+    for (auto path_entry : dlock_buffer) {
```

Comment:
Doesn't this result in extra copies? Should this be const auto & ?

Reply:
There are lots of full copies in this code, but, this is for an I_S table used for debugging deadlocks, so performance isn't necessarily the focus. This was taken directly from the initial upstream implementation. I'll send an upstream PR for these so we can try to keep merging nightmares to a minimum.